crates/wdk-build/src/utils.rs

```diff
@@ -374,39 +398,154 @@ fn read_registry_key_string_value(
 ///   subdirectories
 ///
 /// # Returns
-/// * `Some((major, minor))` - The maximum version found as a tuple of (major,
-///   minor)
+/// * `Some(BasicVersion)` - The maximum version found
 /// * `None` - If no valid version directories are found or if the directory
 ///   cannot be read
-///
-/// # Examples
-/// Given a directory structure:
-/// ```text
-/// /some/path/
-/// ├── 1.4/
-/// ├── 1.5/
-/// └── 1.6/
-/// ```
-/// This function will return `Some((1, 6))`
-pub fn find_max_version_in_directory<P: AsRef<Path>>(directory_path: P) -> Option<(u32, u32)> {
+pub(crate) fn find_max_version_in_directory<P: AsRef<Path>>(
+    directory_path: P,
+) -> Option<BasicVersion> {
     std::fs::read_dir(directory_path.as_ref())
         .ok()?
         .flatten()
-        .filter_map(|entry| {
-            let dir_name = entry.file_name();
-            let dir_name = dir_name.to_str()?;
-            let (major_str, minor_str) = dir_name.split_once('.')?;
-            let major = major_str.parse::<u32>().ok()?;
-            let minor = minor_str.parse::<u32>().ok()?;
-            Some((major, minor))
-        })
+        .filter_map(|entry| BasicVersion::parse(entry.file_name().to_str()?))
         .max()
 }
 
 #[cfg(test)]
 mod tests {
+    use assert_fs::prelude::*;
+
     use super::*;
 
+    mod basic_version {
+        use super::*;
+
+        #[test]
+        fn test_new() {
+            let version = BasicVersion::new(1, 2);
+            assert_eq!(version.major, 1);
+            assert_eq!(version.minor, 2);
+        }
+
+        #[test]
+        fn test_parse_valid_versions() {
```

Comment:
No need to have the prefix `test_` in every test name. It does add anything that we don't already know.